File: models/Schedule.js
import mongoose from 'mongoose';
const { Schema, model } = mongoose;

const ScheduleSchema = new mongoose.Schema({
    ///doctorId: { type: Schema.Types.ObjectId, ref: 'user' },
    dayOfWeek: {
      type: String,
    },
    startTime: {
      type: String,
    },
    endTime: {
      type: String,
    },
    user: {
      type: Schema.Types.ObjectId, 
      ref: 'user' 
 },
 appointement: { type: Schema.Types.ObjectId, ref: 'appointement' },
},
{
    timestamp: true
}
);
export default mongoose.model('schedule', ScheduleSchema);

File: lib/models/schedule.dart
class Schedule {
  String? dayOfWeek;
  String? startTime;
  String? endTime;
  String? user;

  Schedule(
    this.dayOfWeek,
    this.startTime,
    this.endTime,
    this.user,
  );

  Schedule.scheduleWithThreeParameters(
      {this.dayOfWeek, this.startTime, this.endTime});
}
